packages/dotcom-ui-data-embed/src/client/DataEmbedStore.ts

```diff
@@ -0,0 +1,15 @@
+export default class DataEmbedStore {
+  private data
+
+  constructor(data) {
+    this.data = Object.freeze(data)
+  }
+
+  get(key: string) {
+    return this.data.hasOwnProperty(key) ? this.data[key] : undefined
```

Comment:
Wouldn't `this.data[key]` return `undefined` if the key does not exist as a property on `this.data`? I.e. is the conditional check necessary here?

Reply:
I'll admit a lot of this code is me looking for patterns in `flags` and `app-context` and consolidating the logic.

So while I didn't make the decision, I can only assume it's to avoid you being able to try to 'get' properties built into the Object.prototype e.g. `constructor` or `__proto__`.

So I'm not to sure, but perhaps could be simplified alongside `flags` and `app-context` later on?

https://masteringjs.io/tutorials/fundamentals/hasownproperty

Reply:
There is always the assumption of good faith that there is a good reason behind any pre-existing code. Also happy to leave it as is for now.

Reply:
You're right, definitely should have checked this, thanks for the prompt. :)